### ISSUE TITLE
Unable to import more than 11 records at a time using csv file

### ISSUE THREAD

**Issue by pranav70** [UniversalDataTool/universal-data-tool#366]
I am unable to import more than 11 records at a time when I am using a CSV file with a single column as present in the file attached (convert excel to CSV for use) herewith. It always adds first 11 records from the file. It is very shocking to see an open-source project with 1.2k+ stars having such a naive issue/bug.
Note: This file contains 14 rows and 1 header row and having a total number of lines 33.

[Test.xlsx](https://github.com/UniversalDataTool/universal-data-tool/files/5474114/Test.xlsx)

**Comment by seveibar** [comment]
What are you using to import the file? Which dialog?

**Comment by seveibar** [comment]
@pranav70 it seems like you have a xlsx file, were you using the CSV import?

**Comment by pranav70** [comment]
No, I was using the very same file but CSV of the same and I used it on my
windows desktop application, online and docker. Everywhere issue is same.
You can also convert the xlsx file in to csv using Microsoft Excel and use
it.

I used text snippets to import as well and it worked but it didn't used
quote character and imported all the 33 lines as a record instead of 14
records.

Also, what is the input format for Json to import because I wasn't able to
import a list of dictionary as json or a single json file not even a jsonl
file.

**Comment by hysios** [comment]
Test.xlsx is a badly format in my excel, below is screenshot in my pc
![image](https://user-images.githubusercontent.com/103227/97884646-4b762800-1d61-11eb-8873-900a19d0ddf7.png)

**Comment by pranav70** [comment]
Hi,

This is the sample file that I was using. You can use this file for your
use just remember that a CSV file can have multiple multiline
records within quote characters just like this one. Here "text" is the
header in this file.

I also requested you to give me an input format for JSON file apart from
.udt.json format. Does this support JSONL, or list of dictionaries as JSON
or a simple JSON file.

I hope you will resolve this issue soon.

*Thanks and Regards,*
*Pranav*

**Comment by seveibar** [comment]
@pranav70 for me to help you, you'll need to provide the CSV file you're using here. I don't have Microsoft Excel and don't use Windows. It seems @hysios wasn't able to open it- so you may have an invalid file. 

The CSV needs to be RFC4180 compatible, which is the standard we use (and most CSV parsing libraries AFAIK)

As for JSON, check https://github.com/UniversalDataTool/udt-format

JSONL is not currently supported, but is planned.

**Comment by pranav70** [comment]
Here is the file that you can use.

[Test.zip](https://github.com/UniversalDataTool/universal-data-tool/files/5476451/Test.zip)

**Comment by seveibar** [comment]
I think importing a CSV could be more clear (maybe that should be the new issue title). The header column in this case must be "document" (it's instead set to "text" or something), I've corrected the CSV and attached. After adjusting the header everything works as expected. (The txt file I'm uploading is actually a CSV, CSV evidently isn't supported by github)


[TestWithCorrectedHeader.txt](https://github.com/UniversalDataTool/universal-data-tool/files/5476877/TestWithCorrectedHeader.txt)

![Screen Capture_select-area_20201102123505](https://user-images.githubusercontent.com/1910070/97899991-0eca1280-1d08-11eb-8c46-d059e3ed20bd.png)

![Screen Capture_select-area_20201102123454](https://user-images.githubusercontent.com/1910070/97900006-14275d00-1d08-11eb-99ea-c5e6f90da7f6.png)

**Comment by seveibar** [comment]
I'm able to reproduce the bug. There are 15 rows in the CSV as shown below, but the UDT only displays 11 samples after importing.

![Screen Capture_select-area_20201102125401](https://user-images.githubusercontent.com/1910070/97901780-a0d31a80-1d0a-11eb-8590-f9d6c944ae33.png)

![image](https://user-images.githubusercontent.com/1910070/97901821-adf00980-1d0a-11eb-9eea-af497fcd94b0.png)

**Comment by pranav70** [comment]
> I'm able to reproduce the bug. There are 15 rows in the CSV as shown below, but the UDT only displays 11 samples after importing.
> 
> ![Screen Capture_select-area_20201102125401](https://user-images.githubusercontent.com/1910070/97901780-a0d31a80-1d0a-11eb-8590-f9d6c944ae33.png)
> 
> ![image](https://user-images.githubusercontent.com/1910070/97901821-adf00980-1d0a-11eb-9eea-af497fcd94b0.png)

Yes, that's exacty my issue. Also, the header should be "document" is nowhere documented in the docs.

**Comment by seveibar** [comment]
Hey @hysios :) wanna take a stab at the fix? It's been a while since I gave you a video update shout out!

**Comment by seveibar** [comment]
Could be a parameter we're passing to the CSV parsing library or something

**Comment by pranav70** [comment]
If I want to tag "@yourrightscamp" in the text which is like "@ajplus @Kaepernick7 @yourrightscamp", then you can't tag it properly but you can tag "7 @yourrightscamp". You can tag I think it is due to improper tokenization as you are considering continous special characters as a single token.

![tokenization_bug](https://user-images.githubusercontent.com/22179855/97903065-7f9f1800-1d64-11eb-9bb9-9a132becfaff.PNG)

![bug](https://user-images.githubusercontent.com/22179855/97903071-80d04500-1d64-11eb-94ab-105ebe9c9731.PNG)

**Comment by seveibar** [comment]
@pranav70 yep you need custom tokenization for that. Issue #342 is about custom tokenization being provided as an interface parameter. Are you interested in code contribution? Both of these issues are good introductory issues if you are. Otherwise I'm guessing someone will get them soon.

**Comment by pranav70** [comment]
> @pranav70 yep you need custom tokenization for that. Issue #342 is about custom tokenization being provided as an interface parameter. Are you interested in code contribution? Both of these issues are good introductory issues if you are. Otherwise I'm guessing someone will get them soon.

Yes, absolutely I would love to contribute. I have good experience in regular expressions that I can help you with. I have other issues/suggestions to improve this product that we can discuss further.

**Comment by seveibar** [comment]
Feel free to hit me up on [Slack](https://join.slack.com/t/universaldatatool/shared_invite/zt-d8teykwi-iOSOUfxugKR~M4AJN6VL3g)! There's also a [contribution guide](https://github.com/UniversalDataTool/universal-data-tool/blob/master/CONTRIBUTING.md). We're trying to get a lot better at NLP so any help is appreciated :)

**Comment by Codyk12** [comment]
@seveibar Did this ever get resolved? I am running into an issue of only uploading 11 of 500 rows from my csv that is just numbers as such:

sample
1
2
3
4
5
... etc

I don't think I have come out clear on this thread of what is wrong and how to fix it, any help would be great!
